### ISSUE TITLE
plugin inputs.mqtt_consumer Regression 1.24.0 -> 1.24.4 -  configuration specified the fields ...  but they weren't used

### ISSUE THREAD

**Issue by flohoff** [influxdata/telegraf#12472]
### Relevant telegraf.conf

```toml
[[inputs.mqtt_consumer]]
        servers = ["tcp://127.0.0.1:1883"]
        topics = ["tele/+/ISENSOR"]
        data_format = "json_v2"
        [[inputs.mqtt_consumer.topic_parsing]]
                topic = "tele/+/+"
                tags = "_/input/retpolicy"
                retpolicy = "isensor"
        [[inputs.mqtt_consumer.json_v2]]
                [[inputs.mqtt_consumer.json_v2.object]]
                        path = "SML"
                        tags = [ "smartmeter" ]
                        optional = true
                [[inputs.mqtt_consumer.json_v2.object]]
                        path = "ENERGY"
                        tags = [ "powerplug" ]
                        optional = true

[[outputs.influxdb]]
  urls = ["http://127.0.0.1:8086"]
  database = "telegraf"
  retention_policy_tag = "retpolicy"
```

### Logs from Telegraf

```text
root@nuc:/etc/telegraf/telegraf.d# telegraf  --config /etc/telegraf/telegraf.conf --config-directory /etc/telegraf/telegraf.d/ --test --debug
2023-01-09T15:50:34Z W! DeprecationWarning: Value "netsnmp" for option "snmp_translator" of plugin "agent" deprecated since version 1.25.0 and will be removed in 2.0.0: Use 'gosmi' instead
2023-01-09T15:50:34Z E! [telegraf] Error running agent: error loading config file /etc/telegraf/telegraf.d/input-mqtt.conf: plugin inputs.mqtt_consumer: line 1: configuration specified the fields ["retpolicy"], but they weren't used
```

### System info

Debian 11 Telegraf 1.25.0-1

### Steps to reproduce

This is a regression. It used to work with telegraf 1.24.0 and passed this variable to the influx output plugin to control the retention policy based on the MQTT topic. 

It fails with 1.24.4

### Expected behavior

Pass retpolicy to influx output plugin

### Actual behavior

Config parse error

**Comment by powersj** [comment]
> configuration specified the fields ["retpolicy"], but they weren't used

I do not see `retpolicy` in the codebase or in any PR for the mqtt_consumer plugin. The [TopicParsingConfig](https://github.com/influxdata/telegraf/blob/master/plugins/inputs/mqtt_consumer/mqtt_consumer.go#L51) struct has no field either.

> retention_policy_tag = "retpolicy"

This option in the influxdb v2 output plugin specifies a tag to look for and uses that tag's value as the retention policy for a write.

It does look like this error started showing up after v1.24.2 in v1.24.3. Going through the [diff](https://github.com/influxdata/telegraf/compare/v1.24.2...v1.24.3) I came across https://github.com/influxdata/telegraf/commit/59860eb7b194e14a2d5f52fae9d04071a4f00eae which is what re-introduced that error. In fact, if you go back to v1.23.4 you will get the same error message.

Long story short, telegraf should throw an error on unused config values because these are useless and mean that something you put in is not actually working when you may think it is. This behavior was missing/absent in v1.24.0-v1.24.2 and fixed in v1.24.3.
